MVP Module/Tests/Feed UI/FeedUIIntegrationTests.swift

```diff
@@ -280,6 +280,35 @@ final class FeedUIIntegrationTests: XCTestCase {
 		wait(for: [exp], timeout: 1.0)
 	}
 
+	func test_loadFeedCompletion_rendersErrorMessageOnFail() {
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+		loader.completeFeedLoadingWithError()
+
+		XCTAssertEqual(sut.errorMessage, localized("FEED_VIEW_CONNECTION_ERROR"))
+	}
+
+	func test_loadFeedCompletion_hideErrorMessageOnReload() {
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+		loader.completeFeedLoadingWithError()
+		sut.simulateUserInitiatedFeedReload()
+
+		XCTAssertNil(sut.errorMessage)
+	}
+
+	func test_loadFeedCompletion_dismissErrorMessageOnTap() {
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+		loader.completeFeedLoadingWithError()
+		sut.errorView.button.simulateTap()
```

Comment:
The integration test shouldn't be coupled with the view hierarchy such as accessing buttons directly. Instead, create a helper method such as `sut.simulateTapOnError()`

```suggestion
		sut.simulateTapOnError()
```